### ISSUE TITLE
CIS 1.x vs 2.x

### ISSUE THREAD

**Issue by btopro** [elmsln/elmsln#87]
Every CIS instance prior to 0.1.0 release will be known as CIS 1.x. The following will be applied to 1.x prior to 0.1.0:
* Role / permissions fixes (Role fixes in there, permissions still needed)
* Features re-exported to account for it (already happened)
* Ulmus refactor (already happened)

1.x as was expressed in previous community threads, had a marketing component. Rick's group also added in student hub type of stuff. As per the discussion though, these use-cases will all be abstracted in 2.x. The focus of 2.x is purely a data-driving system that only faculty, staff and admins interact with.

2.x major differences as part of abstraction:
* The front facing components can now be found in a better form in the Publicize distribution -- https://www.drupal.org/project/publicize
* The Hub components RIck's group supplied will be backported / re-architected into #76 project inbox.
* 2.x will also get the look and feel of the rest of the network.

The system setup to support both (old and new CIS versions) will be this: Repo proper points to 1.x (so all those currently running things don't have a breaking change). During install, 2.x will be pointed to dynamically. This way all NEW deployments will start on 2.x. All existing will have an upgrade path / bash script that they can run to convert CIS from 1.x to 2.x though it will be something that requires some testing prior to running, especially if you've done custom development.

Basically the change over isn't THAT major (even if it looks like it), all the properties of CIS (course / section backend) will remain the same, it's just a lot of the content types will be removed and the system scope / interfaces simplified.  The focus of this system will be purely on creating and setting things up as well as longer term management. It will also provide a backend to data and help stitch the experience together without people realizing its there (except those who are paid to care about logistical things like that!)

**Comment by btopro** [comment]
It should also be noted that DSLM (which everything is built on top of) makes supporting multiple versions of the same project pretty easy from a developer perspective and there are major versions as folder names all over the under the hood setup for things like this :)

**Comment by btopro** [comment]
2.x initial commit / branch pushed up, not hooked up to anything / tested but at least the place holders are there

**Comment by btopro** [comment]
since I added drush support to https://www.drupal.org/project/profile_switcher for another project, a drush recipe (like the one included via https://www.drupal.org/node/2456491 ) could be used as an optional upgrade routine to get people off of CIS 1.x and onto CIS 2.x with relative ease.

I will be needing to engage in something like this shortly. Current plan is to leave the views largely in tact, make no real changes to the data model (at this time) but gut the bloated permissions handling and make the UX work w/ the new interface flow. We've got the contextual awareness we need via other commits, and I have a use-case for doing a conversion to a live environment (very quickly) so I'm hoping this can get cleaned up by monday.

**Comment by btopro** [comment]
Boom, working! w/ upgrade script that worked in production!

**Comment by btopro** [comment]
as #102  references, we've got a bit of an issue currently w/ hard resets on the git repo, the common method of managing the repo being pushed out to ensure it is correct.

Possible resolution:
Create a 0.0.0 release named something else that's assumed to be running CIS 1.x
After that release point in time, everything else is starting with CIS 2.x and the code repo will reflect that.
Otherwise whenever you do a hard reset, CIS will revert back to what's in the repo which is 1.x which borks the theme and all the features just enabled to make it work :)

**Comment by btopro** [comment]
as I couldn't sleep last night, a competing idea would be to remove the symlink from version control and ignore it via .gitignore. In this scenario we would be relying on the installation script itself to build the stack's association correctly and we'd sidestep the issue referenced in the previous comment.

**Comment by btopro** [comment]
symlink approach saved my vagrant instance's cis symlink By effectively removing this link from core and then building it dynamically (and ignoring that it exists) we're able to allow people to maintain their current instances without any of the proposed hack-ey solutions to do so.

Still UX work to do in 2.x but this is resolved
